### PR TITLE
Added missing code for displaying icons for marked topics

### PR DESCRIPTION
**Pull request by Poffen**
The code to display the proper icons for topics marked as sticky or closed is missing. Also, when a topic is marked as both sticky and closed, both labels (Sticky: & Closed:) are displayed. I suggest that the "Closed" icon/label together with the sticky background are displayed for topics that are marked as both sticky and closed.

### CODE REVIEW

File: featherbb/Model/Forum.php [hunk 1]
```diff
@@ -319,11 +326,17 @@ public function display_topics_moderate($fid, $sort_by, $start_from)
...
+                    $cur_topic['icon_type'] = 'icon icon-sticky';
                     $status_text[] = '<span class="stickytext">'.__('Sticky').'</span>';
+                    // When marked as both sticky and closed, display only the sticky background
+                } elseif ($cur_topic['sticky'] == '1' && $cur_topic['closed'] == '1') {
+                    $cur_topic['item_status'] .= ' isticky';
+                    $cur_topic['icon_type'] = '';
+                    $status_text[] = '';
                 }
-
+
```
Comment (adaur) [inline featherbb/Model/Forum.php]: Could you please remove this? Thanks :)

File: featherbb/Model/Forum.php [hunk 2]
```diff
@@ -216,6 +222,7 @@ public function print_topics($forum_id, $sort_by, $start_from)
                     $cur_topic['subject_formatted'] = '<a href="'.Router::pathFor('Topic', ['id' => $cur_topic['id'], 'name' => $url_subject]).'">'.Utils::escape($cur_topic['subject']).'</a> <span class="byuser">'.__('by').' '.Utils::escape($cur_topic['poster']).'</span>';
                     $status_text[] = '<span class="closedtext">'.__('Closed').'</span>';
                     $cur_topic['item_status'] .= ' iclosed';
+                    $cur_topic['icon_type'] = 'icon icon-closed';
```
Comment (adaur) [inline featherbb/Model/Forum.php:225]: Do you think a moved topic should be flagged as closed, even if it's open in the forum it's been moved in?

File: featherbb/Model/Forum.php [hunk 3]
```diff
@@ -201,9 +201,15 @@ public function print_topics($forum_id, $sort_by, $start_from)
                     $cur_topic['subject'] = Utils::censor($cur_topic['subject']);
                 }
 
-                if ($cur_topic['sticky'] == '1') {
+                if ($cur_topic['sticky'] == '1' && $cur_topic['closed'] == '0') {
                     $cur_topic['item_status'] .= ' isticky';
```
Comment (adaur) [inline featherbb/Model/Forum.php]: Maybe you could refactor the code using a big if ($cur_topic['sticky'] == '1') having the $cur_topic['item_status'] line and do the conditions below?

**Comment by adaur** [comment]
I'm curious, why would you remove the Sticky info if the topic is closed? I think it is useful for the user to know that the topic is both closed and sticky.

Could you provide a screenshot to see what it looks like? Thank you.

**Comment by Poffen** [comment]
Hi adaur,

I should say that I am by no means an expert coder, so if the logic I've used here is inappropriate than I apologize for this. I did however test this in every way I could, and to me it works as intended. A stickied or closed or stickied/closed or unflagged topic should IMO remain so also after it has been moved. And it does so when I test this at my end. At least I couldn't see any side effects. I did notice though that a couple of lines of code may be omitted (the empty array elements).

The idea of keeping just the "Closed" label together with the sticky green background if a topic is both closed and stickied was that this was most intuitive to me. The topic would have (with this template) a green background (telling that it is stickied) but a label and icon which tells that it is closed (see below). It will also appear in the top together with the other stickied topics. I thought this was the best compromise since you cannot have them both (two sets of icons, text-color and background). You could of course use more if statements and include a new label (i.e. "Sticked and closed:") as an alternative. But then again you would have to decide if the label should be colored stickied (green) or closed (red). But I don't think it's a good idea to combine the existing labels (i.e. Sticky: Closed:), with both having a trailing colon. It looks cluttered, I think, although they can be displayed in different colours.

If you think that there is a better way to do this, then feel free to suggest what I should do. You're the expert. :)

skjermbilde fra 2016-05-18 23 02 11

Putte

**Comment by Poffen** [comment]
Hi adaur,

Would this be better, you think? I've included a new language definition for the sticky/closed topic colored as "stickytekst".
new-sticky-closed

Putte

**Comment by adaur** [comment]
What about something like Sticky and closed: ?

**Comment by Poffen** [comment]
Yes, "Sticky and closed:" will be fine. Let me know what color you prefer, "stickytext" (green) or "closedtext" (red), and then I'll refactor the code lines according to your previous suggestion and update my commit later today.

**Comment by adaur** [comment]
I'd prefer the sticky, but I don't have a strong preference

**Comment by Poffen** [comment]
Included another issue I just found. The pagination links on the forum-moderator page are broken, due to missing input in the URL construct.

Putte

**Comment by adaur** [comment]
Thanks for the commits. I just tested the pagination links in the moderator forum without any issue.

**Pull request merged**